astropy/cosmology/parameter.py

```diff
@@ -46,33 +46,41 @@ class Parameter:
 
     _registry_validators = {}
 
-    def __init__(self, fvalidate="default", doc=None, *,
-                 unit=None, equivalencies=[], fmt=".3g", derived=False):
+    _properties = ("name", "derived", "unit", "equivalencies", "format_spec", "fvalidate")
+    """List of property."""
+
+    def __init__(self, *, derived=False, unit=None, equivalencies=[],
+                 fvalidate="default", fmt=".3g", doc=None):
+
+        # attribute name on container cosmology class.
+        # really set in __set_name__, but if Parameter is not init'ed as a
+        # descriptor this ensures that the attributes exist.
+        self._attr_name = self._attr_name_private = None
+
+        self._derived = derived
+        self._fmt = str(fmt)  # @property is `format_spec`
+        self.__doc__ = doc
+
+        # units stuff
+        self._unit = u.Unit(unit) if unit is not None else None
+        self._equivalencies = equivalencies
+
         # parse registered fvalidate
         if callable(fvalidate):
-            pass
+            self._validate_repr = repr(fvalidate)
         elif fvalidate in self._registry_validators:
+            self._validate_repr = fvalidate  # don't need repr()
```

Comment:
I don't think you need this if you use `_init_arguments` in `__repr__`

Reply:
I need to record the string name before processing it to a function, otherwise the ``repr`` looks pretty bad and can't be round-tripped by ``eval``.  ``_init_arguments`` actually requires this as well.

Reply:
I meant that all `_init_arguments` needs is that you store the input somewhere. But I see I made my comment at the wrong point: no need to store `repr(fvalidate)` for the callable case.

Reply:
Agreed. I'm making it `None`, because I prefer an attribute to exist in all cases.
If Python ever allows ``__slots__ = ("__doc__", ...)`` then `Parameter` can be given ``__slots__``, which would be nice.